app/src/main/java/com/example/bottomnav/IntegerDecoder.java

```diff
@@ -0,0 +1,41 @@
+package com.example.bottomnav;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class IntegerDecoder<T> implements DataDecoder {
+    protected int packetSize;
+    protected String payloadDataType;
+    protected T rawValue;
+    protected String value;
+
+    public IntegerDecoder(int byteSize, String dataType) {
+        payloadDataType = dataType;
+        packetSize = byteSize;
+    }
+
+    @Override
+    public String decode(Integer canId, byte[] payload) {
+        ByteBuffer bb = ByteBuffer.wrap(payload);
+        byte[] packet = new byte[packetSize];
+        bb.get(packet, 0, packetSize);
+        rawValue = (T) new Integer(new BigInteger(packet).intValue() & 0xff);
```

Comment:
Using mutable state is highly not recommended, because now calling valueToString requires calling decode right before - a requirement that isn't obvious or intuitive.

I recommend delegating instead:
```java
public String decode(...) {  // maybe rename decodeToString or something?
  return payloadDataType + ": " + valueToRaw(canId, payload)
}

public T valueToRaw(...) {  // would need to take canId and payload
  // actual decoding happens here
}
```

Also, is this somewhat duplicated w/ valueToString? Consider getting rid of one of them - and leaving it to the next level (caller) if it wants to do some specialty formatting. Try to keep your classes as single responsibility as possible.